### PR TITLE
better documentation for MultiThreadedEventLoopGroup

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

MultiThreadedEventLoopGroup's docs didn't make it clear that you leak
threads if not shut down.

Modifications:

Add clarification that you should shut down
MultiThreadedEventLoopGroups.

Result:

hopefully clearer documentation.

### CODE REVIEW

File: Sources/NIO/EventLoop.swift [hunk 1]
```diff
@@ -823,6 +823,18 @@ extension EventLoopGroup {
...
+/// all run their own `EventLoop`. Those threads will not be shut down until `shutdownGracefully` or
+/// `syncShutdownGracefully` is called.
+///
+/// - note: It's good style to call `MultiThreadedEventLoopGroup.shutdownGracefully` or
+///         `MultiThreadedEventLoopGroup.syncShutdownGracefully` when you no longer need this `EventLoopGroup`. In
+///         many cases that is just before your program exits.
+/// - warning: Unit tests often spawn one `MultithreadedEventLoopGroup` per unit test to force isolation between the
+///            tests. In those cases it's important to shut the `MultithreadedEventLoopGroup` down at the end of the
+///            test. A good place to start a `MultithreadedEventLoopGroup` is the `setUp` method of your `XCTestCase`
+///            subclass, a good place to shut it down is the `tearDown` method.
```
Comment (Lukasa) [inline Sources/NIO/EventLoop.swift]: Nit: a number of the references to the name of this class in this block have incorrect casing. The correct casing is `MultiThreadedEventLoopGroup`. Mind fixing those up?

File: Sources/NIO/EventLoop.swift [hunk 2]
```diff
@@ -920,7 +937,14 @@ final public class MultiThreadedEventLoopGroup: EventLoopGroup {
         }
     }
 
-    public func shutdownGracefully(queue: DispatchQueue, _ callback: @escaping (Error?) -> Void) {
+    /// Shut this `MultithreadedEventLoopGroup` down which causes the `EventLoop`s and their associated threads to be
```
Comment (Lukasa) [inline Sources/NIO/EventLoop.swift]: Nit: you have the same casing problem here.

File: Sources/NIO/EventLoop.swift [hunk 3]
```diff
@@ -920,7 +937,14 @@ final public class MultiThreadedEventLoopGroup: EventLoopGroup {
...
+    /// shut down and release their resources.
+    ///
+    /// - parameters:
+    ///    - queue: The `DispatchQueue` to run `handler` on when the shutdown operation finished.
```
Comment (Lukasa) [inline Sources/NIO/EventLoop.swift]: Nit: s/finished/completes/

File: Sources/NIO/EventLoop.swift [hunk 4]
```diff
@@ -920,7 +937,14 @@ final public class MultiThreadedEventLoopGroup: EventLoopGroup {
...
+    ///    - handler: The handler which is called after the shutdown operation finished. The parameter will be `nil`
```
Comment (Lukasa) [inline Sources/NIO/EventLoop.swift]: Nit: s/finished/completes/

**Comment by weissi** [comment]
@Lukasa mind having a quick look again?

**Pull request merged**